### DISCUSSION TITLE
Help with using DotNetCorePublisher with the configurator

### DISCUSSION THREAD

**Discussion by nick5454** [cake-build/cake#3491] [Q&A] [answered]
I slightly confused on using the [DotNetCore.Publish ](https://cakebuild.net/api/Cake.Common.Tools.DotNetCore.Publish/) as it doesn't seem to be the same as MSBuild 

I need to do some with properties to add a "PublishProfile" property to the call. Does that runner support  the code flow like MSBuild below? 

Here's the syntax below that i'm after `dotnet publish WebApplication.csproj /p:PublishProfile=<FolderProfileName>
`
`MSBuild(BuildParameters.SolutionFile.ToString(), configurator =>
			configurator
				.SetConfiguration(BuildParameters.Configuration)
				.WithProperty("Platform", BuildParameters.Platform)`

**Comment by augustoproiete** [comment, marked as answer]
The `dotnet publish` command doesn't expose an argument to pass in a publish profile, so [you have to pass it as a property the same way you would do with MSBuild](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish#msbuild). e.g.:

```powershell
dotnet publish -p:PublishProfile=profile
```

With Cake you would do the same using the [`DotNetCorePublish`](https://cakebuild.net/api/Cake.Common.Tools.DotNetCore/DotNetCoreAliases/A4C0DB33) alias including the `PublishProfile` (or `PublishProfileFullPath` if using  custom path) via an MSBuild property using `WithProperty`.

```cakescript
DotNetCorePublish("./src/Project/Project.csproj", new DotNetCorePublishSettings
{
    // ...

    MSBuildSettings = new DotNetCoreMSBuildSettings()
        .WithProperty("PublishProfile", folderProfile)
    });
});
```

**Reply by nick5454** [reply]
Completely missed that, Thanks!
